openfl-tutorials/interactive_api/PyTorch_DogsCats_ViT/envoy/dogs_cats_shard_descriptor.py

```diff
@@ -0,0 +1,188 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Cats and dogs shard descriptor."""
+
+import json
+import os
+import shutil
+from hashlib import md5
+from pathlib import Path
+from random import shuffle
+from zipfile import ZipFile
+
+import numpy as np
+from kaggle.api.kaggle_api_extended import KaggleApi
+from PIL import Image
+
+from openfl.interface.interactive_api.shard_descriptor import ShardDataset
+from openfl.interface.interactive_api.shard_descriptor import ShardDescriptor
+
+
+class DogsCatsShardDataset(ShardDataset):
+    """Dogs and cats Shard dataset class."""
+
+    def __init__(self, data_type: str, dataset_dir: Path,
+                 rank=1, worldsize=1, enforce_image_hw=None):
+        """Initialize DogsCatsShardDataset."""
+        self.rank = rank
+        self.worldsize = worldsize
+        self.dataset_dir = dataset_dir
+        self.enforce_image_hw = enforce_image_hw
+        self.img_path = self.dataset_dir / data_type
+
+        self.img_names = [
+            img.name
+            for img in sorted(self.img_path.iterdir())
+            if img.suffix == '.jpg'
+        ]
+
+        # Sharding
+        self.img_names = self.img_names[self.rank - 1::self.worldsize]
+        # Shuffling the results dataset after choose half pictures of each class
+        shuffle(self.img_names)
+
+    def __getitem__(self, index):
+        """Return a item by the index."""
+        name = self.img_names[index]
+        # Reading data
+        img = Image.open(self.img_path / name)
+        img_class = 1 if name[:3] == 'dog' else 0
```

Comment:
Just a little suggestion
```suggestion
        img_class = 1 if name.startswith('dog') else 0
```